File: typescript-version/src/family/father.ts
import { Parent } from './parent';
import { Gender } from './person';
import { Mother } from './mother';

/**
 * Represents a father in the family
 */
export class Father extends Parent {
  // x genome and y genome
  public genomeX: string;
  public genomeY: string;

  constructor(name: string) {
    super(name, Gender.MALE);
    this.genomeX = "x";
    this.genomeY = "y";
  }

  /**
   * Marry a mother
   */
  marry(spouse: Mother): void {
    super.connect(spouse);
  }

  /**
   * Get the genome based on gender
   */
  getGenome(gender: Gender): string {
    return gender === Gender.FEMALE ? this.genomeX : this.genomeY;
  }

  /**
   * Describe the father's genome
   */
  describeGenome(): string {
    return 'X, Y';
  }
}

File: python-version/family/father.py
"""
Father class - represents a father in the family
"""

from .parent import Parent
from .person import Gender
import random


class Father(Parent):
    """Represents a father in the family"""
    
    def __init__(self, name: str):
        """Initialize a father"""
        super().__init__(name, Gender.MALE)
        # x genome and y genome
        self.genome_x = "x"
        self.genome_y = "y"

    def marry(self, mother: 'Mother') -> None:
        """
        Marry a mother
        """
        super().connect(mother)

    def get_genome(self, gender: Gender) -> str:
        """
        Get the genome based on gender
        """
        return self.genome_x if gender == Gender.FEMALE else self.genome_y

    def describe_genome(self) -> str:
        """
        Describe the father's genome
        """
        return 'X, Y'
